WFLY-4835 remove references to ClassIndex

stuartwdouglas opened the pull request.

stuartwdouglas committed:
WFLY-4835 remove references to ClassIndex

jamezp commented on ee/src/main/java/org/jboss/as/ee/component/ComponentConfiguration.java:
```diff
@@ -94,7 +94,7 @@
     private final Set<Method> lifecycleMethods = new HashSet<>();
 
-    public ComponentConfiguration(final ComponentDescription componentDescription, final ClassIndex classIndex, final ClassLoader moduleClassLoader, final ModuleLoader moduleLoader) {
+    public ComponentConfiguration(final ComponentDescription componentDescription, final ClassReflectionIndex classIndex, final ClassLoader moduleClassLoader, final ModuleLoader moduleLoader) {
```
Should this be `ClassReflectionIndex<?>` for consistency?

jamezp commented on ee/src/main/java/org/jboss/as/ee/component/ComponentConfiguration.java:
```diff
@@ -505,7 +505,7 @@ public void setInstanceFactory(final ComponentFactory instanceFactory) {
     }
 
-    public ClassIndex getClassIndex() {
+    public ClassReflectionIndex getClassIndex() {
```
Should this be `ClassReflectionIndex<?>` for consistency?

stuartwdouglas replied:
Really ClassReflectionIndex should not be parameterised. I will remove all the <?> from full, so I can make the change in core.

stuartwdouglas committed:
Remove type paramaters from ClassReflectionIndex

This type only referes to deployment classes, the type paramter will never be a known type

jamezp commented:
Checkstyle :(

stuartwdouglas commented:
retest this please

stuartwdouglas commented:
Retest this please
